Assets/Scripts/Player/PlayerInventory.cs

```diff
@@ -52,10 +57,12 @@ public override void Awake()
 	// Use this for initialization
 	void Start()
 	{
-		inventoryBar = GameObject.FindObjectOfType<InventoryScrollBar>();
-		DebugUtils.Assert( inventoryBar, "There must be an InventoryScrollBar object in the scene." );
+		_inventoryBar = GameObject.FindObjectOfType<InventoryScrollBar>();
+		DebugUtils.Assert( _inventoryBar, "There must be an InventoryScrollBar object in the scene." );
+
+		DayCycleManager.RegisterEndOfDayCallback( HideBackBuddy );
```

Comment:
This way makes more sense to me because it's not possible to not feed a buddy in a day, and not all buddies needs to be re-enabled at the end of each day.